src/needs-parens.js

```diff
@@ -38,10 +38,7 @@ function needsParens(path) {
   }
 
   // Avoid parens in short control structures like `if (expr) statement`
-  if (
-    ["if", "while", "for", "foreach"].includes(parent.kind) &&
-    (parent.body === node || parent.alternate === node)
```

Comment:
Not sure which case this was protecting, maybe you know more @evilebottnawi?

Reply:
@czosel to avoid parens around
```php
if ($var = 1) ($var = 1);
```

Just uncomment this check and update snapshots and you can see what happens :smile:

Reply:
Strange, I don't see that change. In which test file is it?

Reply:
Also notice in the changes below that no existing test is changed.

Reply:
@czosel strange, give me time to investigate

Reply:
@czosel rebase PR and try to update snapshot for `parens` directory, you can see how it is break some cases